cmd/deploy/deploy.go

```diff
@@ -266,10 +266,24 @@ func (dc *DeployCommand) RunDeploy(ctx context.Context, deployOptions *Options)
 		if deployOptions.Manifest != nil {
 			deployOptions.Manifest.Name = deployOptions.Name
 		}
-		if deployOptions.Manifest.Deploy != nil && deployOptions.Manifest.Deploy.Compose != nil && deployOptions.Manifest.Deploy.Compose.Stack != nil {
-			deployOptions.Manifest.Deploy.Compose.Stack.Name = deployOptions.Name
+		if deployOptions.Manifest.Deploy != nil && deployOptions.Manifest.Deploy.ComposeSection != nil && deployOptions.Manifest.Deploy.ComposeSection.Stack != nil {
+			deployOptions.Manifest.Deploy.ComposeSection.Stack.Name = deployOptions.Name
 		}
 	}
+
+	if deployOptions.Manifest != nil && deployOptions.Manifest.Deploy != nil && deployOptions.Manifest.Deploy.ComposeSection != nil {
```

Comment:
We could include this if clause in the one above. If a composeSection has any manifest defined it must have a Stack field not nil